package/MDAnalysis/coordinates/XDR.py

```diff
@@ -289,12 +290,16 @@ def _read_frame(self, i):
         return timestep
 
     def _read_next_timestep(self, ts=None):
+        # NOTE: TRR implements its own version
         """copy next frame into timestep"""
         if self._frame == self.n_frames - 1:
             raise IOError(errno.EIO, 'trying to go over trajectory limit')
         if ts is None:
             ts = self.ts
-        frame = self._xdr.read()
+        if ts.has_positions:
+            frame = self._xdr.read_direct(ts.positions)
+        else:
+            frame = self._xdr.read()
```

Comment:
If TRR implements its own then why have a generic method here that only works for XTC?

Let's remove `_read_next_timestep` from the base class. Then we can also cleanly remove `read_direct` from TRR.